exercises/binary-search/src/example/java/BinarySearch.java

```diff
@@ -0,0 +1,40 @@
+
+import java.util.List;
+
+public class BinarySearch<T extends Comparable<T>> {
```

Comment:
Just noticed that this generic type means the example implementation is a lot more general than the tests require. @exercism/java is this a good feature in a reference solution, or does it veer towards YAGNI territory?